File: ssTools/netcat/php/nc.php
<?php

$server = stream_socket_server("tcp://0.0.0.0:6969", $errno, $errstr);

if (!$server) {
    die("Gagal membuka soket: $errstr ($errno)");
}

echo "Menjalankan server netcat di port 6969...\n";

while (true) {
    $client = stream_socket_accept($server);

    $clientAddress = stream_socket_get_name($client, true);

    echo "Menerima koneksi dari $clientAddress\n";

    while (!feof($client)) {
        $data = fread($client, 8192);
        echo $data;
    }

    fclose($client);
    echo "Koneksi ditutup\n\n";
}

fclose($server);
?>

File: ssTools/netcat/java/nc.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.ServerSocket;
import java.net.Socket;

public class nc {

    public static void main(String[] args) {
        try {
            ServerSocket serverSocket = new ServerSocket(6969);
            System.out.println("Menjalankan server netcat di port 6969...");

            while (true) {
                Socket clientSocket = serverSocket.accept();
                System.out.println("Menerima koneksi dari " + clientSocket.getInetAddress().getHostAddress());

                BufferedReader reader = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
                String line;
                while ((line = reader.readLine()) != null) {
                    System.out.println(line);
                }

                clientSocket.close();
                System.out.println("Koneksi ditutup\n");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}